### PR TITLE
Topology optimization improvements

### PR DESCRIPTION
**Pull request by pcarruscag**
## Proposed Changes
Couple efficiency/functionality improvements to the topology optimization implementation as detailed in TestCases/fea_topology/config.cfg

Edit: This requires c++11 so it will have to go after the PR that enables it.

## PR Checklist
- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [X] My contribution is commented and consistent with SU2 style.
- [X] I have added a test case that demonstrates my contribution, if necessary.

### CODE REVIEW

File: TestCases/fea_topology/quick_start/settings_refnode.cfg [hunk 1]
```diff
@@ -3,6 +3,9 @@ PHYSICAL_PROBLEM= ELASTICITY
 MATH_PROBLEM= DISCRETE_ADJOINT
 DYNAMIC_ANALYSIS= NO
 %
+SINGLEZONE_DRIVER=YES
+ITER= 1
```
Comment (pcarruscag) [inline TestCases/fea_topology/quick_start/settings_refnode.cfg]: @rsanfer , @talbring can you guys please explain the current maximum iteration stopping criteria in an issue?

Comment (talbring) [reply TestCases/fea_topology/quick_start/settings_refnode.cfg]: With this we already went a little bit ahead of ourselves. We thought we would make progress a little bit faster. I am currently preparing a documentation for controlling the solver, where I will explain all of this. For the singlezone adjoint solver it is simply EXT_ITER -> ITER. Everything else is still the same.

File: Common/src/geometry_structure.cpp [hunk 2]
```diff
@@ -3337,11 +3336,11 @@ void CGeometry::FilterValuesAtElementCG(const vector<su2double> filter_radius,
         /*--- morphology kernels (image processing) ---*/
         case DILATE_MORPH_FILTER: case ERODE_MORPH_FILTER:
             
-          for (vector<long>::iterator it=neighbours.begin(); it!=neighbours.end(); ++it)
+          for (auto idx : neighbours)
```
Comment (talbring) [inline Common/src/geometry_structure.cpp]: We have to be careful not to overuse the `auto` keyword. It does not always help with readability.

**Comment by pcarruscag** [comment]
Thanks for the reviews guys. At the moment the optimization framework does not support the topology variables, there is a script in SU2_PY for this feature that runs a testcase with hardcoded parameters...
It is not an easy thing to run a tutorial for :/

**Pull request merged**